obs-packaging/gen_versions_txt.sh

```diff
@@ -24,24 +24,36 @@ get_kata_version() {
 	curl -SsL "https://raw.githubusercontent.com/${project}/runtime/${branch}/VERSION"
 }
 
+get_kata_hash() {
+	repo=$1
+	ref=$2
+	git ls-remote --heads --tags "https://github.com/${project}/${repo}.git" | grep "${ref}" | awk '{print $1}'
+}
+
 gen_version_file() {
 	local branch="$1"
-	[ -n "${branch}" ] || exit 1
+	local use_head="$2"
+	local ref="refs/heads/${branch}"
+	local kata_version="${branch}"
+
+	if [ -n "$branch" ] && [ -z "${use_head}" ]; then
+		kata_version=$(get_kata_version "${branch}")
+		ref="refs/tags/${kata_version}^{}"
+	fi
 
-	local kata_version=$(get_kata_version "${branch}")
-	kata_runtime_hash=$(get_kata_hash_from_tag "runtime" "${kata_version}")
-	kata_proxy_hash=$(get_kata_hash_from_tag "proxy" "${kata_version}")
-	kata_shim_hash=$(get_kata_hash_from_tag "shim" "${kata_version}")
-	kata_agent_hash=$(get_kata_hash_from_tag "agent" "${kata_version}")
-	kata_ksm_throttler_hash=$(get_kata_hash_from_tag "ksm-throttler" "${kata_version}")
+	kata_runtime_hash=$(get_kata_hash "runtime" "${ref}")
+	kata_proxy_hash=$(get_kata_hash "proxy" "${ref}")
+	kata_shim_hash=$(get_kata_hash "shim" "${ref}")
+	kata_agent_hash=$(get_kata_hash "agent" "${ref}")
+	kata_ksm_throttler_hash=$(get_kata_hash "ksm-throttler" "${ref}")
 
 	qemu_lite_branch=$(get_from_kata_deps "assets.hypervisor.qemu-lite.branch" "${kata_version}")
 	qemu_lite_version=$(curl -s -L "https://raw.githubusercontent.com/${project}/qemu/${qemu_lite_branch}/VERSION")
 	qemu_lite_hash=$(git ls-remote https://github.com/${project}/qemu.git | grep "refs/heads/${qemu_lite_branch}" | awk '{print $1}')
 
-	qemu_vanilla_branch=$(get_from_kata_deps "assets.hypervisor.qemu.version" "${kata_version}")
-	qemu_vanilla_version=$(curl -s -L "https://raw.githubusercontent.com/qemu/qemu/${qemu_vanilla_branch}/VERSION")
-	qemu_vanilla_hash=$(git ls-remote https://github.com/qemu/qemu.git | grep "refs/heads/${qemu_vanilla_branch}" | awk '{print $1}')
+	qemu_vanilla_tag=$(get_from_kata_deps "assets.hypervisor.qemu.tag" "${kata_version}")
```

Comment:
Super Nit. But is there a reason we should still have the version and not just use the tag everywhere. Trying to see if we can clean up some cruft from age-old.

Reply:
It depends, we can't use the tag everywhere because sometimes we want to test master (the latest commit) in those cases we don't have a tag